test/e2e/Tests/Tests/SuspendResumeTests.cs

```diff
@@ -0,0 +1,197 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+using System.Net;
+using Xunit;
+using Xunit.Abstractions;
+
+namespace Microsoft.Azure.Durable.Tests.DotnetIsolatedE2E;
+
+[Collection(Constants.FunctionAppCollectionName)]
+public class SuspendResumeTests
+{
+    private readonly FunctionAppFixture _fixture;
+    private readonly ITestOutputHelper _output;
+
+    public SuspendResumeTests(FunctionAppFixture fixture, ITestOutputHelper testOutputHelper)
+    {
+        _fixture = fixture;
+        _fixture.TestLogs.UseTestLogger(testOutputHelper);
+        _output = testOutputHelper;
+    }
+
+
+    [Fact]
+    public async Task SuspendAndResumeRunningOrchestration_ShouldSucceed()
+    {
+        using HttpResponseMessage response = await HttpHelpers.InvokeHttpTrigger("LongOrchestrator_HttpStart", "");
+
+        Assert.Equal(HttpStatusCode.Accepted, response.StatusCode);
+        string instanceId = await DurableHelpers.ParseInstanceIdAsync(response);
+        string statusQueryGetUri = await DurableHelpers.ParseStatusQueryGetUriAsync(response);
+
+        Thread.Sleep(1000);
+
+        var orchestrationDetails = await DurableHelpers.GetRunningOrchestrationDetailsAsync(statusQueryGetUri);
+        Assert.Equal("Running", orchestrationDetails.RuntimeStatus);
```

Comment:
Rather than sleeping and hoping that the status will change after the sleep interval, we can make these checks a bit more reliable and potentially faster if we use a polling pattern to check for the state change. This would address the very common flakey test problem where engineers need to increase sleep durations because things ran slower than we expected. Consider creating a helper method that can wait for an orchestration to transition from one state to another (Pending --> Running, Running --> Suspended, Suspended --> Running, etc.)